org.mwc.debrief.pepys/src/main/java/org/mwc/debrief/pepys/model/ModelConfiguration.java

```diff
@@ -135,7 +135,11 @@ public void addPropertyChangeListener(final PropertyChangeListener l) {
 	@Override
 	public void apply() throws Exception {
 		validate();
-		currentItems = TreeUtils.buildStructure(this);
+		if (ALGORITHM_TYPE.FAST_MODE.equals(getAlgorithmType())) {
```

Comment:
Hello @saulhidalgoaular - I think a `switch` statement would be tidier here - it will check we handle all the possibilities. Right?